RecoMET/METProducers/src/HcalNoiseInfoProducer.cc

```diff
@@ -112,6 +113,49 @@ HcalNoiseInfoProducer::HcalNoiseInfoProducer(const edm::ParameterSet& iConfig) :
     edm::LogWarning("HCalNoiseInfoProducer") << " forcing fillRecHits to be true if fillDigis is true.\n";
   }
 
+  // get the fiber configuration vectors
+  std::vector<int> TmpLaserMonDetTypeList = iConfig.getParameter<std::vector<int> >("LaserMonDetTypeList");
+  std::vector<int> TmpLaserMonIPhiList = iConfig.getParameter<std::vector<int> >("LaserMonIPhiList");
+  std::vector<int> TmpLaserMonIEtaList = iConfig.getParameter<std::vector<int> >("LaserMonIEtaList");
+
+  // the transfer of data from python to c seems to have issues
+  // this can be fixed by explicitly filling the vector
+  for( std::vector<int>::const_iterator itr = TmpLaserMonDetTypeList.begin();
+          itr != TmpLaserMonDetTypeList.end(); ++itr ) {
+      LaserMonDetTypeList_.push_back( *itr );
+  }
+
+  for( std::vector<int>::const_iterator itr = TmpLaserMonIPhiList.begin();
+          itr != TmpLaserMonIPhiList.end(); ++itr ) {
+      LaserMonIPhiList_.push_back( *itr );
+  }
+
+  for( std::vector<int>::const_iterator itr = TmpLaserMonIEtaList.begin();
+          itr != TmpLaserMonIEtaList.end(); ++itr ) {
+      LaserMonIEtaList_.push_back( *itr );
+  }
+
+  // check that the vectors have the same size, if not
+  // disable the laser monitor
+  if( !( (LaserMonDetTypeList_.size() == LaserMonIEtaList_.size() ) && 
+         (LaserMonDetTypeList_.size() == LaserMonIPhiList_.size() ) ) ) { 
+    edm::LogWarning("MisConfiguration")<<"Must provide equally sized lists for LaserMonDetTypeList, LaserMonIEtaList, and LaserMonIPhiList.  Will not fill LaserMon\n";
+    fillLaserMonitor_=false;
+  }
+
+  // get the integration region with defaults
+  if( iConfig.existsAs<int>("LaserMonitorTSStart" ) ) {
```

Comment:
use of existsAs should be justified, because it is an exception.
Default values for parameters should be defined with fillDescriptions.
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp